Add FAQ: update a field depending on another field (#281)
eclipsesource/jsonforms2-website: content/faq/faq.mdx

  "default": "Max"
}
```

## How to update a field based on another field in JSON Forms?

Consider a carwash application offering various services and calculating the resulting price.
The schema includes a multiselect field called services and a price attribute, with the price field set as readonly.

<DependentFieldExample/>

There are two approaches to update a field in JSON Forms based on the value of another field:

#### Approach 1: Implementing functionality in the onChange method of JSON Forms

In this approach, you can implement the logic in the onChange method of JSON Forms. 

> The onChange method triggers whenever a user changes the form data.

Reviewer: Mention that this approach should only be used when using an older JSON Forms without middleware support as it works in a similar way, however the middleware variant is more performant.

```js
export const CarWash = () => {
  const [formData, setFormData] = useState(inputData);
  
  const onChange = ({ data, _errors }) => {
    const price = data.services.length*15;
    if (data.price === price){
        setFormData(data);
    }else{
        setFormData({...data, price: price})
    }
  }

  return (
    <JsonForms
      data={formData}
      schema={schema}
      renderers={renderers}
      onChange={onChange}
    />
  );
};
```

In the onChange method, the price is calculated based on the selected services. 
If the calculated price matches the existing value, the data remains unchanged.
However, if the price differs, a new data object is created with the updated price and passed to JSON Forms.

> It's important to only create a new object when necessary to avoid infinite loops.

Reviewer: This sentence should be in a warning element. We already have some in the website.

<DependentFieldExample/>

There are three approaches to update a field in JSON Forms based on the value of another field: 
Utilizeing the JSONF Forms middleware, using the onChange method or create a custom render.

#### Approach 1: JSON Forms middleware

We can utilize the JSON Forms middleware to compute and set the price.
JSON Forms utilizes the reducer pattern and various actions to update its state.
The middleware intercepts the call to the JSON Forms reducers and calls your custom code instead. 
For detailed insights into the JSON Forms middleware, the reducer pattern, and JSON Forms actions, refer to the documentation [here](/docs/middleware). 
In this scenario, we want to customize the behavior associated with the  `UPDATE_DATA` action, which is triggered when the form's data is changed by the user.
We initially invoke JSON Forms default reducer to update the data and identify any errors.
Subsequently, we adjust the price fields based on the selected services and update the state with the newly calculated data.
We additionally override the `INIT` and `UPDATE_CORE` actions, in case the data prop passed to JSON Forms doesn't have the correct price set yet.

```js

> import { INIT, UPDATE_DATA } from  '@jsonforms/core'

Reviewer: ```suggestion
import { INIT, UPDATE_CORE, UPDATE_DATA } from  '@jsonforms/core'
```

...
const middleware = useCallback((state, action, defaultReducer) => {
  const newState = defaultReducer(state, action);
  switch (action.type) {
    case INIT:
    case UPDATE_CORE:
    case UPDATE_DATA: {
      if (newState.data.services.length * 15 !== newState.data.price) {
        newState.data.price = newState.data.services.length * 15;
      }
      return newState;
    }
    default:
      return newState;
  }
});

...

<JsonForms
  data={data}
  schema={schema}
  renderers={materialRenderers}
  middleware={middleware}
/>
```

#### Approach 2: Implementing functionality in the onChange method of JSON Forms

In this approach, you can implement the logic in the onChange method of JSON Forms. 
The onChange method triggers whenever a user changes the form data.

:::info Note

> Its more performant to use the middlewear approach, but this is a good approach for JSON Forms before version v3.2.0

Reviewer: ```suggestion
It's more performant to use the middleware approach, but this is a good approach for JSON Forms before version v3.2.0
```

:::

```js
export const CarWash = () => {
  const [formData, setFormData] = useState(inputData);
  
  const onChange = ({ data, _errors }) => {
    const price = data.services.length*15;
    if (data.price === price){
        setFormData(data);

>     }else{

Reviewer: ```suggestion
    } else {
```
